src/com/rockbite/bootcamp/Product.java

```diff
@@ -0,0 +1,65 @@
+package com.rockbite.bootcamp;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Product which must be sold in the shop
+ */
+public class Product {
+
+    //unique identifier
+    private int id;
+
+    //category of the product
+    private Category category;
+
+    //payload of the product
+    private Map<Integer, Item> payload = new HashMap<>();
```

Comment:
Why is this a map instead of an array?
Furthermore, this maps are just being accessed via getters, instead of having smart methods like "add item" remove item and so on. As a result adding this getters is basically the same as just making this maps public.